index.js

```diff
@@ -122,15 +122,17 @@ const onChange = (object, onChange, options = {}) => {
 			if (!inApply) {
 				inApply = true;
 
+				const path = pathCache.get(target);
+				const previous = options.clone ? options.clone(thisArg) : undefined;
 				const result = Reflect.apply(target, thisArg, argumentsList);
 
+				inApply = false;
+
 				if (changed) {
-					onChange();
+					handleChange(path.substring(0, path.lastIndexOf('.')), '', previous, thisArg);
```

Comment:
Can you use `slice` instead of `substring`?

See https://github.com/sindresorhus/eslint-plugin-unicorn/issues/65